proxy/http/HttpTransact.cc

```diff
@@ -6487,9 +6487,15 @@ HttpTransact::is_request_valid(State *s, HTTPHdr *incoming_request)
 bool
 HttpTransact::is_request_retryable(State *s)
 {
+  // If safe requests are  retryable, it should be safe to retry safe requests irrespective of bytes sent or connection state
+  // according to RFC the following methods are safe (https://tools.ietf.org/html/rfc7231#section-4.2.1)
   // If there was no error establishing the connection (and we sent bytes)-- we cannot retry
-  if (s->current.state != CONNECTION_ERROR && s->state_machine->server_request_hdr_bytes > 0 &&
-      s->state_machine->get_server_session()->get_netvc()->outstanding() != s->state_machine->server_request_hdr_bytes) {
+  if (!(s->txn_conf->safe_requests_retryable == 1 &&
```

Comment:
Nit picky, but just treat is as a boolean? I.e. why explicitly compare it with "1" ? We generally don't do that for other similar configs.